microprofile/server/src/main/java/io/helidon/microprofile/server/ServerCdiExtension.java

```diff
@@ -582,4 +594,94 @@ private HttpRouting.Builder findRouting(String className,
 
         return serverNamedRoutingBuilder(routingName);
     }
+
+    // -- CDI bean producers for ServerRequest and ServerResponse -----------------------
+
+    private void afterBeanDiscovery(@Observes AfterBeanDiscovery event) {
+        event.addBean(new ServerRequestProducer());
+        event.addBean(new ServerResponseProducer());
+    }
+
+    /**
+     * Base class for server producers sharing common methods.
+     */
+    static class ServerProducer {
+
+        @SuppressWarnings("all")
+        public static class DefaultAnnotationLiteral extends AnnotationLiteral<Default> implements Default {
```

Comment:
Were you aware of https://jakarta.ee/specifications/cdi/3.0/apidocs/jakarta/enterprise/inject/default.literal?

Reply:
No, using it now.